go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -64,13 +64,14 @@ func doDoltCheckout(ctx *sql.Context, args []string) (int, error) {
 		return 1, errors.New("Improper usage.")
 	}
 
-	// Checking out new branch.
 	dSess := dsess.DSessFromSess(ctx.Session)
-	dbData, ok := dSess.GetDbData(ctx, dbName)
+	// dbData should use the current database data, which can be at revision database.
+	dbData, ok := dSess.GetDbData(ctx, revDbName)
```

Comment:
Ahh!! I see the bug now. Even for when we were using a revision db, we always only grabbed the DbData for the main database. Makes sense. 

I think the `revDbName` could be a little clearer though... instead of calling it `revDbName` (which sometimes it will be a revDb, sometimes it won't be) maybe there's something better like `currentDatabase` that would be a little clearer?